Multiline: Auto-align Grids

radian628 opened the pull request:
I will let the image explain:

![image](https://github.com/DesModder/DesModder/assets/70352880/1dc30cae-3ce0-40f9-8cce-dff4148b0be7)

radian628 commented:
![image](https://github.com/DesModder/DesModder/assets/70352880/b88a1439-3f02-4638-b272-e335a287d2da)
vertical align looks alright too

radian628 commented:
Fixes #759

jared-hughes commented on localization/en.ftl:
```diff
@@ -275,16 +275,21 @@ multiline-name = Multiline Expressions
 multiline-opt-widthBeforeMultiline-desc = Minimum width (as a percent of the viewport size) at which point wrapping will occur. On mobile, this value is tripled.
-multiline-opt-automaticallyMultilinify-name = Insert Linebreaks while Typing
-multiline-opt-automaticallyMultilinify-desc = Automatically splits expressions onto multiple lines while you type, bypassing the need to use Ctrl+M.
+multiline-opt-alterLayoutWhileTyping-name = Alter Layout while Typing
```
"Update" would be a little clearer than "Alter", here and below

radian628 replied:
Def agreed. Will change.

jared-hughes commented:
Cursor has a bunch of space around it:
![image](https://github.com/DesModder/DesModder/assets/20214911/5f357e70-0131-407e-ad6c-95a76cf3b1ed)

Paste in `L=\left[1,2,3,4,\ \ \ 5,6,7,8,9,25,\ \ \ 10,11,12,13,\ \ \ 14,15,16,17,18,19\right]` then delete the `25,`. Bad state:

![image](https://github.com/DesModder/DesModder/assets/20214911/c95db314-a932-42e9-93bb-cf19e3b4416c)

radian628 commented:
> Cursor has a bunch of space around it:

Yep, def a bug.

> Paste in `L=\left[1,2,3,4,\ \ \ 5,6,7,8,9,25,\ \ \ 10,11,12,13,\ \ \ 14,15,16,17,18,19\right]` then delete the `25,`. Bad state:

Also a bug. Seems like it isn't refreshing readily enough.

radian628 commented:
I think I figured out what's going on with this: It's treating the `dcg-mq-cursor` element as part of the equation. Getting rid of that solves the issue of the cursor leading to misalignment. Now the bigger misalignment bug (the `25,` one) seems to only occur when there's something other than a comma at the end of a line. On one hand this is an obvious visual indicator that something is missing. On the other hand it looks rather ugly and will cause a lot of layout changes as a user types, so I'm inclined to fix it.

radian628 commented:
Alright, both of these should be fixed now.

radian628 commented:
Another change I might make: some kind of warning if a number or variable name or something along those lines has a line break in the middle of it, possibly indicating an accidental missing comma. I could probably use a red underline or something or the other to indicate this, something like this: 
![image](https://github.com/DesModder/DesModder/assets/70352880/a375aa87-c33f-4651-b2af-66a2f63d7ea8)

Notice how 1314 is in this list even though the intent is probably to have just 13 and 14.

radian628 commented:
![image](https://github.com/DesModder/DesModder/assets/70352880/882f117d-369b-43e8-b271-684efa6c4653)
Added the red underline and fixed a weird bug causing the width to balloon to thousands of pixels.

jared-hughes commented:
I love the red underline!

Though it seems it doesn't work in two cases that are common. When there's no overhang, and when the line ends in a variable: 
![image](https://github.com/DesModder/DesModder/assets/20214911/6a3da015-9aa4-49cd-b222-3c056835105b)

<https://www.desmos.com/calculator/vhv0cmwv1j>
